src/app/code/community/Omise/Gateway/Model/Payment/Base/Payment.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+
+class Omise_Gateway_Model_Payment_Base_Payment extends Omise_Gateway_Model_Payment {
+    /**
+     * @var string
+     */
+    protected $_code;
+
+    /**
+     * @var string
+     */
+    protected $_formBlockType;
+
+    /**
+     * @var string
+     */
+    protected $_infoBlockType;
+
+    /**
+     * Payment Method features
+     *
+     * @var bool
+     */
+    protected $_isGateway;
+
+    /**
+     * Payment Method features
+     *
+     * @var bool
+     */
+    protected $_canReviewPayment;
+
+    /**
+     * Payment Method features
+     *
+     * @var bool
+     */
+    protected $_isInitializeNeeded;
+
+    /**
+     * @var string
+     */
+    protected $_callbackUrl;
+
+    /**
+     * @param string $code
+     */
+    public function setCode($code)
+    {
+        $this->_code = $code;
+    }
+
+    /**
+     * @param string $formBlockType
+     */
+    public function setFormBlockType($formBlockType)
+    {
+        $this->_formBlockType = $formBlockType;
+    }
+
+    /**
+     * @param string $infoBlockType
+     */
+    public function setInfoBlockType($infoBlockType)
+    {
+        $this->_infoBlockType = $infoBlockType;
+    }
+
+    /**
+     * @param bool $isGateway
+     */
+    public function setIsGateway($isGateway)
+    {
+        $this->_isGateway = $isGateway;
+    }
+
+    /**
+     * @param bool $canReviewPayment
+     */
+    public function setCanReviewPayment($canReviewPayment)
+    {
+        $this->_canReviewPayment = $canReviewPayment;
+    }
+
+    /**
+     * @param bool $isInitializeNeeded
+     */
+    public function setIsInitializeNeeded($isInitializeNeeded)
+    {
+        $this->_isInitializeNeeded = $isInitializeNeeded;
+    }
+
+    /**
+     * @param string $callbackUrl
+     */
+    public function setCallbackUrl($callbackUrl)
+    {
+        $this->_callbackUrl = $callbackUrl;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCallbackUrl()
+    {
+        return $this->_callbackUrl;
+    }
+
+    public function __construct()
```

Comment:
We can remove this - if it isn't defined, the parent constructor will be called automatically